### ISSUE TITLE
HTTP/3: Incomplete Data Transfer When Used with Spring Boot WebFlux

### ISSUE THREAD

**Issue by murphye** [eclipse/jetty.project#10537]
**Jetty version(s)**
Jetty 12.0.1

**Jetty Environment**
Embedded Spring Boot WebFlux

**Java version/vendor** `(use: java -version)`
openjdk version "20.0.1" 2023-04-18

**OS type/version**
macOS 13.5.2

**Description**
This is a follow-on issue to https://github.com/eclipse/jetty.project/issues/10519, but a completely separate problem related to using Jetty HTTP/3 with Spring WebFlux.

I have created a separate branch for WebFlux here: https://github.com/murphye/spring-boot-http-3-jetty/tree/reactive

When the client attempts to use HTTP/3, the response is interrupted. Here is what Curl and Firefox state:

Curl: ` transfer closed with 27 bytes remaining to read`
Firefox: `NS_ERROR_NET_PARTIAL_TRANSFER`

There is some conflict with `HTTP3ServerConnector` running in the context of Spring WebFlux where a HTTP/3 request cannot complete successfully.

**How to reproduce?**

1. Clone `reactive` branch to your local machine: https://github.com/murphye/spring-boot-http-3-jetty/tree/reactive
2. Follow the `README.md` to run the application
3. Observe the stated errors in Firefox and `curl` when making a request as HTTP/3

**Comment by murphye** [comment]
Note that Jetty is the server used by Spring WebFlux in this case. Here is how that's configured with Maven:

```xml
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-webflux</artifactId>
			<exclusions>
				<exclusion>
					<groupId>org.springframework.boot</groupId>
					<artifactId>spring-boot-starter-reactor-netty</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-jetty</artifactId>
		</dependency>
```

**Comment by lorban** [comment]
@murphye A fix for this problem just got merged into the jetty-12.0.x branch. Could you please confirm that it works for you by either building the branch yourself (or waiting until tomorrow for the nightly build to be published in the [Jetty snapshot repo](https://oss.sonatype.org/content/repositories/jetty-snapshots)) and report back here?

Thanks!
